File: backend/src/Tmap.Api/Features/Subtasks/SubtaskDtos.cs
namespace Tmap.Api.Features.Subtasks;

public sealed record CreateSubtaskRequest(string Title);

public sealed record UpdateSubtaskRequest(string? Title, bool? Completed, int? SortOrder);

public sealed record SubtaskResponse(
    Guid Id,
    Guid TaskId,
    string Title,
    bool Completed,
    int SortOrder,
    DateTimeOffset CreatedAt,
    DateTimeOffset UpdatedAt);

File: android/app/src/main/java/net/qmindtech/tmap/data/remote/dto/SubtaskDtos.kt
package net.qmindtech.tmap.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class CreateSubtaskRequest(
    val id: String? = null,
    val title: String,
)

@Serializable
data class UpdateSubtaskRequest(
    val title: String? = null,
    val completed: Boolean? = null,
    val sortOrder: Int? = null,
)

@Serializable
data class SubtaskResponse(
    val id: String,
    val taskId: String,
    val title: String,
    val completed: Boolean,
    val sortOrder: Int,
    val createdAt: String,
    val updatedAt: String,
)
